http-server-netty/src/main/java/io/micronaut/http/server/netty/handler/Compressor.java

```diff
@@ -0,0 +1,188 @@
+        DeflateOptions stdDeflate = StandardCompressionOptions.deflate();
+        this.deflateOptions = StandardCompressionOptions.deflate(strategy.getCompressionLevel(), stdDeflate.windowBits(), stdDeflate.memLevel());
+        this.zstdOptions = Zstd.isAvailable() ? StandardCompressionOptions.zstd() : null;
+        this.snappyOptions = StandardCompressionOptions.snappy();
+    }
+
+    @Nullable
+    ChannelHandler prepare(HttpRequest request, HttpResponse response) {
+        // from HttpContentEncoder: isPassthru
+        int code = response.status().code();
+        if (code < 200 || code == 204 || code == 304 ||
+            (request.method().equals(HttpMethod.HEAD) || (request.method().equals(HttpMethod.CONNECT) && code == 200)) ||
+            response.protocolVersion() == HttpVersion.HTTP_1_0) {
+            return null;
+        }
+        // special case for FHR to keep behavior identical to HttpContentEncoder
+        if (response instanceof FullHttpResponse fhr && !fhr.content().isReadable()) {
+            return null;
+        }
+        if (!strategy.shouldCompress(response)) {
+            return null;
+        }
+        if (response.headers().contains(HttpHeaderNames.CONTENT_ENCODING)) {
+            // already encoded
+            return null;
+        }
+        List<String> acceptEncoding = new ArrayList<>();
+        for (String s : request.headers().getAll(HttpHeaderNames.ACCEPT_ENCODING)) {
+            acceptEncoding.addAll(Arrays.asList(s.split(",")));
+        }
+        Algorithm encoding = determineEncoding(acceptEncoding);
+        if (encoding == null) {
+            return null;
+        }
+        response.headers().add(HttpHeaderNames.CONTENT_ENCODING, encoding.contentEncoding);
+        return switch (encoding) {
+            case BR -> makeBrotliEncoder();
+            case ZSTD -> new ZstdEncoder(zstdOptions.compressionLevel(), zstdOptions.blockSize(), zstdOptions.maxEncodeSize());
+            case SNAPPY -> new SnappyFrameEncoder();
+            case GZIP -> ZlibCodecFactory.newZlibEncoder(ZlibWrapper.GZIP, gzipOptions.compressionLevel(), gzipOptions.windowBits(), gzipOptions.memLevel());
+            case DEFLATE -> ZlibCodecFactory.newZlibEncoder(ZlibWrapper.ZLIB, deflateOptions.compressionLevel(), deflateOptions.windowBits(), deflateOptions.memLevel());
+        };
+    }
+
+    private BrotliEncoder makeBrotliEncoder() {
+        return new BrotliEncoder(brotliOptions.parameters());
+    }
+
+    @SuppressWarnings("FloatingPointEquality")
+    private Algorithm determineEncoding(List<String> acceptEncoding) {
+        // from HttpContentCompressor, slightly modified
+        float starQ = -1.0f;
+        float brQ = -1.0f;
+        float zstdQ = -1.0f;
+        float snappyQ = -1.0f;
+        float gzipQ = -1.0f;
+        float deflateQ = -1.0f;
+        for (String encoding : acceptEncoding) {
```

Comment:
I wonder if this can be refactored as a `record(encoding, q, Algorithm) implements Comparable` and use `stream().sort().first()`

Reply:
It's copied straight from netty so i dont want to change it

Reply:
easier to port future changes that way :)

Reply:
Ok